src/internal/ppsutil/util.go

```diff
@@ -46,19 +46,18 @@ import (
 	"github.com/pachyderm/pachyderm/v2/src/server/worker/common"
 )
 
-// PipelineRepo creates a pfs repo for a given pipeline.
-func PipelineRepo(pipeline *pps.Pipeline) *pfs.Repo {
-	return client.NewRepo(pipeline.Name)
-}
-
 // PipelineRcName generates the name of the k8s replication controller that
 // manages a pipeline's workers
-func PipelineRcName(name string, version uint64) string {
+func PipelineRcName(projectName, pipelineName string, version uint64) string {
 	// k8s won't allow RC names that contain upper-case letters
 	// or underscores
 	// TODO: deal with name collision
-	name = strings.ReplaceAll(name, "_", "-")
-	return fmt.Sprintf("pipeline-%s-v%d", strings.ToLower(name), version)
+	pipelineName = strings.ReplaceAll(pipelineName, "_", "-")
+	if projectName == "" {
+		return fmt.Sprintf("pipeline-%s-v%d", strings.ToLower(pipelineName), version)
+	}
+	projectName = strings.ReplaceAll(projectName, "_", "-")
+	return fmt.Sprintf("pipeline-%s-%s-v%d", strings.ToLower(projectName), strings.ToLower(pipelineName), version)
```

Comment:
we should test the limits of name length here. My main concern is that an existing pipeline rc name and this change will make the RC creation fail when pachyderm is updated

Reply:
The limit is 253 characters. Project and pipeline names are limited to 63 characters and `pipeline---v` is 12 characters, so that leaves 253 - (63 + 63 + 12) = 118 characters free for the decimal version number.